Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs

```diff
@@ -1,21 +1,21 @@
 
 namespace StreetCode.DAL.Repositories.Interfaces.Base;
 
-public interface IRepositoryBase
+public interface IRepositoryBase<T>
```

Comment:
As a side note, during the latter stages of our development plan we would surely face the need insome form of conditional fetching of only one entity from the repo.
Hence, the base repo interface must be supplied with FirstOrDefaultAsync/SingleOrDefaultAsync methods.